src/handler.py

```diff
@@ -272,11 +293,15 @@ def on_cap(cli, svr, mynick, cmd, *caps):
     if botconfig.SASL_AUTHENTICATION:
         @hook("authenticate")
         def auth_plus(cli, something, plus):
+            nonlocal selected_sasl
```

Comment:
It looks like you're just using `selected_sasl`'s value and not changing it, so `nonlocal` isn't required here.